components/squad/wikis/smite/squad_custom.lua

```diff
@@ -0,0 +1,168 @@
+		cell:node(mw.html.create('div'):addClass('MobileStuff'):wikitext('Position:&nbsp;'))
+
+		if String.isNotEmpty(args.position) then
+			cell:wikitext(args.position)
+			if String.isNotEmpty(args.role) then
+				cell:wikitext('&nbsp;(' .. args.role .. ')')
+			end
+		elseif String.isNotEmpty(args.role) then
+			cell:wikitext(args.role)
+		end
+	end
+
+	self.content:node(cell)
+
+	self.lpdbData.position = args.position
+	self.lpdbData.role = args.role or self.lpdbData.role
+
+	return self
+end
+
+---@param playerList table[]
+---@param squadType integer
+---@return Html?
+function CustomSquad.runAuto(playerList, squadType)
+	if Table.isEmpty(playerList) then
+		return
+	end
+
+	local squad = Squad()
+	squad:init(mw.getCurrentFrame())
+
+	squad.type = squadType
+
+	squad.header = CustomSquad.header
+	squad:title():header()
+
+	for _, player in pairs(playerList) do
+		--Get Reference(s)
+		local joinReference = SquadAutoRefs.useReferences(player.joindateRef, player.joindate)
+		local leaveReference = SquadAutoRefs.useReferences(player.leavedateRef, player.leavedate)
+
+		-- Map between formats
+		player.joindate = (player.joindatedisplay or player.joindate) .. ' ' .. joinReference
+		player.leavedate = (player.leavedatedisplay or player.leavedate) .. ' ' .. leaveReference
+		player.inactivedate = player.leavedate
+
+		player.link = player.page
+		player.role = player.thisTeam.role
+		player.position = player.thisTeam.position
+		player.team = player.thisTeam.role == 'Loan' and player.oldTeam.team
+
+		player.newteam = player.newTeam.team
+		player.newteamrole = player.newTeam.role
+		player.newteamdate = player.newTeam.date
+
+		squad:row(CustomSquad._playerRow(player, squad.type))
+	end
+
+	return squad:create()
+end
+
+---@param player table
+---@param squadType integer
+---@return Html
+function CustomSquad._playerRow(player, squadType)
+	local row = ExtendedSquadRow()
+
+	row:status(squadType)
+	row:id({
+		(player.idleavedate or player.id),
+		flag = player.flag,
+		link = player.link,
+		captain = player.captain,
+		role = player.role,
+		team = player.team,
+		date = player.leavedate or player.inactivedate or player.leavedate,
+	})
+	row:name{name = player.name}
+	row:position{role = player.role, position = player.position}
+	row:date(player.joindate, 'Join Date:&nbsp;', 'joindate')
+
+	if squadType == Squad.SquadType.FORMER then
+		row:date(player.leavedate, 'Leave Date:&nbsp;', 'leavedate')
+		row:newteam{
+			newteam = player.newteam,
+			newteamrole = player.newteamrole,
+			newteamdate = player.newteamdate,
+			leavedate = player.leavedate
+		}
+	elseif squadType == Squad.SquadType.INACTIVE then
+		row:date(player.inactivedate, 'Inactive Date:&nbsp;', 'inactivedate')
+	end
+
+	return row:create(
+		mw.title.getCurrentTitle().prefixedText .. '_' .. player.id .. '_' .. ReferenceCleaner.clean(player.joindate)
+		.. (player.role and '_' .. player.role or '')
+		.. '_' .. squadType
+	)
+end
```

Comment:
This can be significantly simplified without the manual ones

Reply:
![image](https://github.com/Liquipedia/Lua-Modules/assets/88981446/2cfe7f15-d619-4016-8cbc-997782acefe5)
 cant get past this one so far I can only assume that it needs this part below to work 

![image](https://github.com/Liquipedia/Lua-Modules/assets/88981446/e61b4df6-6123-43a8-8c0e-9515f1bf4835)